src/Lean/Elab/Notation.lean

```diff
@@ -43,15 +44,70 @@ def expandNotationItemIntoPattern (stx : Syntax) : MacroM Syntax :=
   else
     Macro.throwUnsupported
 
+partial def removeNonSemanticParenthesis (stx : Syntax) : MacroM Syntax := do
```

Comment:
```suggestion
partial def removeNonSemanticParentheses (stx : Syntax) : MacroM Syntax := do
```
I think it would be OK to drop the `NonSemantic` part. There are different kinds of parentheses, and then there are *the* parentheses.

Reply:
As for implementation, see also https://github.com/leanprover/lean4/blob/b6f251bcd33e3941cd653777068bd0bb2919a6a4/tests/lean/run/Reparen.lean#L14-L20
(the style would be slightly different today)